Libraries/rocSOLVER/syev_batched/main.cpp

```diff
@@ -0,0 +1,264 @@
+#include <rocsolver/rocsolver.h>
+
+#include <hip/hip_runtime.h>
+
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <random>
+#include <string>
+#include <vector>
+
+int main(const int argc, char* argv[])
+{
+    // 1. Parse user input.
+    cli::Parser parser(argc, argv);
+    parser.set_optional<rocblas_int>("n", "n", 3, "Size of n x n input matrix A");
+    parser.set_optional<int>("c", "batch_size", 3, "Number of matrices in the input batch");
+    parser.run_and_exit_if_error();
+
+    // Input sanity checks.
+    const rocblas_int n          = parser.get<rocblas_int>("n");
+    const rocblas_int batch_size = parser.get<rocblas_int>("c");
+
+    if(n <= 0)
+    {
+        std::cout << "Value of 'n' should be greater or equal to 1" << std::endl;
+        return error_exit_code;
+    }
+    if(batch_size <= 0)
+    {
+        std::cout << "Value of 'c' (batch size) should be greater or equal to 1" << std::endl;
+        return error_exit_code;
+    }
+
+    // Calculate the number of elements needed for the batch.
+    const rocblas_int lda                           = n;
+    const rocblas_int single_matrix_num_elements    = lda * n;
+    const rocblas_int batch_eigenvalue_num_elements = n * batch_size;
+
+    // 2. Data vectors.
+    std::vector<rocblas_double> A(single_matrix_num_elements * batch_size); // Input matrix.
+    std::vector<rocblas_double> V(single_matrix_num_elements
+                                  * batch_size); // Resulting eigenvectors.
+    std::vector<rocblas_double> W(batch_eigenvalue_num_elements); // Resulting eigenvalues.
+    std::vector<rocblas_int>    info(batch_size); // Convergence information.
+
+    // 3. Generate a random symmetric matrix.
+    std::default_random_engine                     generator;
+    std::uniform_real_distribution<rocblas_double> distribution(0., 2.);
+    auto random_number = [&]() { return distribution(generator); };
+
+    for(int k = 0; k < batch_size; ++k)
+    {
+        int offset = k * single_matrix_num_elements;
+        for(int i = 0; i < n; ++i)
+        {
+            A[(n + 1) * i + offset] = random_number();
+            for(int j = 0; j < i; ++j)
+            {
+                A[i * n + j + offset] = A[j * n + i + offset] = random_number();
+            }
+        }
+    }
+
+    // 4. Set rocSOLVER parameters.
+    const rocblas_evect evect = rocblas_evect::rocblas_evect_original;
+    const rocblas_fill  uplo  = rocblas_fill::rocblas_fill_lower;
+
+    // 5. Reserve and copy data to device.
+    rocblas_double*  array_A[batch_size];
+    rocblas_double** array_dA;
+    rocblas_double*  d_A    = nullptr;
+    rocblas_double*  d_W    = nullptr;
+    rocblas_int*     d_info = nullptr;
+
+    const rocblas_int info_size  = sizeof(rocblas_int) * batch_size;
+    const rocblas_int array_size = batch_size * sizeof(rocblas_double**);
+
+    const rocblas_int full_size = batch_size * single_matrix_num_elements * sizeof(rocblas_double*);
```

Comment:
Should that be `sizeof(rocblas_double)` rather than `sizeof(rocblas_double*)`?

Reply:
Looks like so. Changed!